torchtext/prototype/models/t5/modules.py

```diff
@@ -830,24 +832,44 @@ def __init__(
             ]
         )
         self.num_layers = num_layers
+        self.norm = T5LayerNorm(d_model)
+        self.dropout1 = nn.Dropout(dropout)
+        self.dropout2 = nn.Dropout(dropout)
 
     def forward(
         self,
-        tgt: Tensor,
+        tgt: Optional[Tensor] = None,
```

Comment:
Similar to the forward of the entire model, if the target is already embedded, no need to inlcude the raw tokenized tgt.